exchanges/kucoin/kucoin_test.go

```diff
@@ -567,115 +356,178 @@ func TestPostOrder(t *testing.T) {
-		t.Errorf("PostMarginOrder() expected %v, but found %v", errInvalidSize, err)
-	}
+	require.ErrorIs(t, err, order.ErrAmountBelowMin)
+
 	sharedtestvalues.SkipTestIfCredentialsUnset(t, ku, canManipulateRealOrders)
 	// default order type is limit and margin mode is cross
-	_, err = ku.PostMarginOrder(context.Background(),
+	result, err := ku.PostMarginOrder(context.Background(),
 		&MarginOrderParam{
 			ClientOrderID: "5bd6e9286d99522a52e458de",
 			Side:          "buy", Symbol: marginTradablePair,
 			Price: 1000, Size: 0.1, PostOnly: true})
-	if err != nil {
-		t.Error("PostMarginOrder() error", err)
-	}
+	require.NoError(t, err)
+	require.NotNil(t, result)
 
 	// market isolated order
-	_, err = ku.PostMarginOrder(context.Background(),
+	result, err = ku.PostMarginOrder(context.Background(),
 		&MarginOrderParam{
 			ClientOrderID: "5bd6e9286d99522a52e458de",
 			Side:          "buy", Symbol: marginTradablePair,
 			OrderType: "market", Funds: 1234,
-			Remark: "remark", MarginMode: "cross", Price: 1000, PostOnly: true, AutoBorrow: true})
-	if err != nil {
-		t.Error("PostMarginOrder() error", err)
-	}
+			Remark: "remark", MarginModel: "cross", Price: 1000, PostOnly: true, AutoBorrow: true})
+	require.NoError(t, err)
+	assert.NotNil(t, result)
+}
+
+func TestPostMarginOrderTest(t *testing.T) {
+	t.Parallel()
+	// default order type is limit
+	_, err := ku.PostMarginOrderTest(context.Background(), &MarginOrderParam{
+		ClientOrderID: ""})
+	require.ErrorIs(t, err, order.ErrClientOrderIDMustBeSet)
+	_, err = ku.PostMarginOrderTest(context.Background(), &MarginOrderParam{
+		ClientOrderID: "5bd6e9286d99522a52e458de", Symbol: marginTradablePair,
+		OrderType: ""})
+	require.ErrorIs(t, err, order.ErrSideIsInvalid)
+	_, err = ku.PostMarginOrderTest(context.Background(), &MarginOrderParam{
+		ClientOrderID: "5bd6e9286d99522a52e458de", Symbol: currency.EMPTYPAIR,
+		Size: 0.1, Side: "buy", Price: 234565})
+	require.ErrorIs(t, err, currency.ErrSymbolStringEmpty)
+	_, err = ku.PostMarginOrderTest(context.Background(), &MarginOrderParam{
+		ClientOrderID: "5bd6e9286d99522a52e458de", Side: "buy",
+		Symbol:    marginTradablePair,
+		OrderType: "limit", Size: 0.1})
+	require.ErrorIs(t, err, order.ErrPriceBelowMin)
+	_, err = ku.PostMarginOrderTest(context.Background(), &MarginOrderParam{
+		ClientOrderID: "5bd6e9286d99522a52e458de", Symbol: marginTradablePair, Side: "buy",
+		OrderType: "limit", Price: 234565})
+	require.ErrorIs(t, err, order.ErrAmountBelowMin)
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, ku)
+	// default order type is limit and margin mode is cross
+	result, err := ku.PostMarginOrderTest(context.Background(),
+		&MarginOrderParam{
+			ClientOrderID: "5bd6e9286d99522a52e458de",
+			Side:          "buy", Symbol: marginTradablePair,
+			Price: 1000, Size: 0.1, PostOnly: true})
+	require.NoError(t, err)
+	require.NotNil(t, result)
+
+	// market isolated order
+	result, err = ku.PostMarginOrderTest(context.Background(),
```

Comment:
I did receive this when running tests:
```
        	Error:      	Received unexpected error:
        	            	failed to retry request, status: 429 Too Many Requests, authenticated request failed
        	Test:       	TestPostMarginOrderTest
```
Would you be able to verify rate limits?